### PR TITLE
Create tests for TimeSlot, extend TimeSlot

### LINKED ISSUE #208: Test for TimeSlot

### PR DESCRIPTION
**Pull request by fbtom**
closes #208

### CODE REVIEW

File: modules/backend/visits/visit/Visit.hpp [hunk 1]
```diff
@@ -14,9 +14,10 @@ class Visit : public iVisit
     // used to set informations required for visit from patient
     void setDoctorInfo(const clinic::staff::doctor::iDoctor& doctor) override;
     void setPatientInfo(const patient::iPatient& patient) override;
-    void setPaymentInfo(const patient::Payments& payments) override;
+    void setPayment(const patient::Payments& payments) override;
     void setRoom(const clinic::room::iRoom& room) override;
     void setTreatmentsInfo(const patient::Treatments& treatments) override;
+    patient::Payments getPayment() const override;
```
Comment (fbtom) [inline modules/backend/visits/visit/Visit.hpp:20]: As Visit blueprint is not finished yet I had to extend it's interface and use to test assignment behaviour in TimeSlot class.

File: modules/backend/calendar/TimeSlot.hpp [hunk 2]
```diff
@@ -10,14 +10,28 @@ class TimeSlot
...
+
+    /* In terms of comparing time_point and duration the following operators are
+      not applicable as there is no use case for them.
+      TimeSlots can be compared if they are equal in multiple WorkUnit objects to
+      compare if there's a possibility to create another TimeSlot object.
+      Timeslotc can be bigger or smaller than other TimeSlot object to check if
+      timepoint of TimeSlot is not overlapping with another TimeSlot object.
+    */
+    bool operator<=(const TimeSlot& other) const = delete;
+    bool operator>=(const TimeSlot& other) const = delete;
```
Comment (BaqWin) [inline modules/backend/calendar/TimeSlot.hpp:34]: I don't see use cases for this operators.

**Review by BaqWin — APPROVED** [review]
I don't se use cases for some operators, but they can be deleted later. Overall code looks good.

**Comment by ziobron** [comment]
Thanks @BaqWin for Code Review.
🏅 1 XP granted

**Comment by ziobron** [comment]
Your PR was merged!
🏅 2 XP granted. Thanks @fbtom for making course materials better!

**Pull request merged**